src/test/java/com/google/cloud/spanner/pgadapter/python/PythonTransactionTests.java

```diff
@@ -0,0 +1,1095 @@
+        2,
+        getWireMessagesOfType(QueryMessage.class).stream()
+            .filter(qm -> qm.toString().contains("READ COMMITTED"))
+            .count());
+  }
+
+  private void restartServerWithDifferentVersion(String version) throws Exception {
+    stopMockSpannerAndPgAdapterServers();
+    doStartMockSpannerAndPgAdapterServers("d", Arrays.asList("-v", version));
+    Assert.assertEquals(version, pgServer.getOptions().getServerVersion());
+  }
+
+  @Test
+  public void testIsolationLevelInTransactions() throws IOException, InterruptedException {
+    // tests isolation_level settings using connection.isolation_level variable
+    List<String> statements = new ArrayList<>();
+    String sql1 = "Select * from some_table";
+    String sql2 = "insert into some_table(col1, col2) values(value1, value2)";
+    String sql3 = "Select * from some_table3";
+    String sql4 = "insert into some_table3(col1, col2) values(value1, value2)";
+
+    statements.add("query");
+    statements.add(sql1);
+
+    statements.add("update");
+    statements.add(sql2);
+
+    statements.add("update");
+    statements.add(sql4);
+
+    statements.add("transaction");
+    statements.add("commit");
+
+    // This will not cause error because the server version is 1.0,
+    // so the psycopg2 will send SERIALIZABLE instead of REPEATABLE READ as the isolation level
+    statements.add("transaction");
+    statements.add("set isolation_level 2");
+
+    statements.add("query");
+    statements.add(sql3);
+
+    statements.add("transaction");
+    statements.add("commit");
+
+    mockSpanner.putStatementResult(
+        StatementResult.query(Statement.of(sql1), createResultSet(1, "abcd")));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(sql2), 1));
+    mockSpanner.putStatementResult(
+        StatementResult.query(Statement.of(sql3), createResultSet(2, "pqrs")));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(sql4), 2));
+
+    String expectedOutput = "(1, 'abcd')\n" + "1\n" + "2\n" + "(2, 'pqrs')\n";
+    String actualOutput = executeTransactions(pgServer.getLocalPort(), statements);
+    Assert.assertEquals(expectedOutput, actualOutput);
+
+    Assert.assertEquals(2, mockSpanner.countRequestsOfType(CommitRequest.class));
+    Assert.assertEquals(0, mockSpanner.countRequestsOfType(RollbackRequest.class));
+    Assert.assertEquals(4, mockSpanner.countRequestsOfType(ExecuteSqlRequest.class));
+
+    List<String> sql = new ArrayList<>();
+
+    for (ExecuteSqlRequest request : mockSpanner.getRequestsOfType(ExecuteSqlRequest.class)) {
+      sql.add(request.getSql());
+    }
+
+    Assert.assertEquals(1, Collections.frequency(sql, sql1));
+    Assert.assertEquals(1, Collections.frequency(sql, sql2));
+    Assert.assertEquals(1, Collections.frequency(sql, sql3));
+    Assert.assertEquals(1, Collections.frequency(sql, sql4));
+
+    mockSpanner.clearRequests();
+    statements.set(9, "set isolation_level 3");
```

Comment:
Try to split these tests into separate test cases. Ideally, each test case should only test one scenario.

Reply:
Done